### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by jchauchat**
Add Smagorinsky model for both fluid and particle phase + TKE budget for LES

### CODE REVIEW

File: TurbulenceModels/turbulenceModels/LES/fluidDynamicLagrangian/fluidDynamicLagrangian.C [hunk 1]
```diff
@@ -40,6 +40,9 @@ void fluidDynamicLagrangian<BasicTurbulenceModel>::correctNut
 {
     this->nut_ = min((flmb_/fmmb_)*sqr(this->delta())*mag(dev(symm(gradU))),
                      nutMax_);
+//    this->nut_ = min((flmb_/max(fmmb_,fmmb0_))*sqr(this->delta())*mag(dev(symm(gradU))),
+//                     nutMax_);
+
```
Comment (CyrilleBonamy) [inline TurbulenceModels/turbulenceModels/LES/fluidDynamicLagrangian/fluidDynamicLagrangian.C:45]: Is it necessary to keep this comment?

File: solver/pU/pEqn.H [hunk 2]
```diff
@@ -216,11 +216,11 @@ License
...
 
-#if (defined(OFVERSION) && (OFVERSION >= 700))
+//#if (defined(OFVERSION) && (OFVERSION >= 700))
             pEqn.solve();
-#else
-            pEqn.solve(mesh.solver(p_rbgh.select(pimple.finalInnerIter())));
-#endif
+//#else
+//            pEqn.solve(mesh.solver(p_rbgh.select(pimple.finalInnerIter())));
+//#endif
```
Comment (CyrilleBonamy) [inline solver/pU/pEqn.H:223]: Why?

File: solver/pUf/pEqn.H [hunk 3]
```diff
@@ -254,11 +254,11 @@ License
...
 
-#if (defined(OFVERSION) && (OFVERSION >= 700))
+//#if (OFVERSION > 700)
             pEqn.solve();
-#else
-            pEqn.solve(mesh.solver(p_rbgh.select(pimple.finalInnerIter())));
-#endif
+//#else
+//            pEqn.solve(mesh.solver(p_rbgh.select(pimple.finalInnerIter())));
+//#endif
```
Comment (CyrilleBonamy) [inline solver/pUf/pEqn.H:261]: again.... why are you deleting those lines necessary to support multiple versions of openfoam?

**Review by CyrilleBonamy — CHANGES_REQUESTED** [review]
In addition to the other comments, I think we need to plan LES micro-testcases for the Continuous Integration.

File: solver/pU/pEqn.H (hunk 2 above)
Comment (jchauchat) [reply solver/pU/pEqn.H:223]: It does not compile with these lines

**Pull request closed without merging**